### PR TITLE
Draft: mark all notifications as read

### LINKED ISSUE #316: Mark all notifications as read
A PR addressing this issue should:

1. Add a button to the top of `/notifications` if there are any unread notifications
2. Clicking the button should mark all unread notifications as read

### PR DESCRIPTION
**Pull request by gathuku**
### Pull request checklist

- [x] Your code contains tests relevant for code you modified
- [ ] You have linted and tested the project with `bin/check`

closes #316 

UI
Screen Shot 2022-03-12 at 12 18 50 PM

### CODE REVIEW

**Comment by gathuku** [comment]
@joemasilotti Getting issues with open startup reporting tests locally. Seems to be passing on the Github CI though.

File: test/integration/read_notifications_test.rb [hunk 1]
```diff
@@ -26,4 +26,22 @@ class ReadNotificationsTest < ActionDispatch::IntegrationTest
...
     assert_select "h1", "Read notifications"
   end
+
+  test "you can read all unread notifications" do
+    user = users(:with_business)
+    developer = developers(:available)
+    sign_in user
+    conversation = Conversation.create!(developer:, business: user.business)
+    ["Hello!", "Available", "Let's talk"].each do |text|
+      Message.create!(conversation:, sender: developer, body: text)
```
Comment (joemasilotti) [inline test/integration/read_notifications_test.rb:36]: Are they any already existing unread notifications from fixtures that we can use?

File: app/controllers/read_notifications_controller.rb [hunk 2]
```diff
@@ -6,4 +6,10 @@ class ReadNotificationsController < ApplicationController
   def index
     @pagy, @notifications = pagy(current_user.notifications.read.newest_first)
   end
+
+  def create
+    current_user.notifications.unread.each(&:mark_as_read!)
```
Comment (joemasilotti) [inline app/controllers/read_notifications_controller.rb:11]:
```suggestion
    current_user.notifications.unread.mark_as_read!
```

**Comment by joemasilotti** [comment]
Weird, can you post the failure here? I'm not seeing any issues when I run locally.

**Comment by gathuku** [comment]
This is an example. could it be something with the stripe setup?
```
Failure:
OpenStartup::ReportingTest#test_refreshes_Expense_records_(grouped_by_month_and_merged_with_additional_fees)_and_adds_manual_expenses [/Users/gathuku/Code/Rails/railsdevs.com/test/models/open_startup/reporting_test.rb:36]
Minitest::Assertion: --- expected
+++ actual
@@ -1 +1 @@
-[[Sat, 01 Jan 2022, "Stripe fees", 0.77e1], [Thu, 30 Dec 2021, "Expense", 0.999e1]]
+[[Sat, 01 Jan 2022, "Stripe fees", 3.3], [Tue, 01 Feb 2022, "Stripe fees", 4.4], [Thu, 30 Dec 2021, "Expense", 9.99]]
```

File: test/integration/read_notifications_test.rb (hunk 1 above)
Comment (gathuku) [reply test/integration/read_notifications_test.rb:36]: Looks like the features are not adding any notifications.

**Comment by joemasilotti** [comment]
Weird! I'm not running into any issues with the test, even after a fresh clone (and no Stripe credentials loaded).

**Comment by gathuku** [comment]
Thank you I will continue exploring what could be the issue. Can't wait to pick another issue.

**Comment by joemasilotti** [comment]
Love it! I'm posting 3 more now – they should be ready to pick up within the hour :)

**Pull request merged**